File: src/System.Data.Odbc/src/Common/System/Data/Common/SafeNativeMethods.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.
// See the LICENSE file in the project root for more information.

using System.Runtime.InteropServices;

namespace System.Data
{
    internal partial class SafeNativeMethods
    {
        internal static IntPtr LocalAlloc(IntPtr initialSize)
        {
            var handle = Marshal.AllocHGlobal(initialSize);
            ZeroMemory(handle, (int)initialSize);
            return handle;
        }

        internal static void LocalFree(IntPtr ptr)
        {
            Marshal.FreeHGlobal(ptr);
        }

        internal static void ZeroMemory(IntPtr ptr, int length)
        {
            var zeroes = new byte[length];
            Marshal.Copy(zeroes, 0, ptr, length);
        }
    }
}

File: src/Microsoft.VisualBasic.Core/src/Microsoft/VisualBasic/Helpers/SafeNativeMethods.vb
' Licensed to the .NET Foundation under one or more agreements.
' The .NET Foundation licenses this file to you under the MIT license.
' See the LICENSE file in the project root for more information.

Imports System.Runtime.InteropServices
Imports System.Runtime.Versioning

Namespace Microsoft.VisualBasic.CompilerServices

    <System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)>
    <ComVisible(False)>
    Friend NotInheritable Class _
        SafeNativeMethods

        <ResourceExposure(ResourceScope.None)>
        <PreserveSig()> Friend Declare Sub _
            GetLocalTime _
                Lib "kernel32" (ByVal systime As NativeTypes.SystemTime)

        '''*************************************************************************
        ''' ;New
        ''' <summary>
        ''' FxCop violation: Avoid uninstantiated internal class. 
        ''' Adding a private constructor to prevent the compiler from generating a default constructor.
        ''' </summary>
        Private Sub New()
        End Sub
    End Class

End Namespace

